cmd/signature-v4.go

```diff
@@ -260,7 +276,7 @@ func doesPresignedSignatureMatch(hashedPayload string, r *http.Request, region s
 	// Construct the query.
 	query.Set(xhttp.AmzDate, t.Format(iso8601Format))
 	query.Set(xhttp.AmzExpires, strconv.Itoa(expireSeconds))
-	query.Set(xhttp.AmzSignedHeaders, getSignedHeaders(extractedSignedHeaders))
+	query.Set(xhttp.AmzSignedHeaders, getSignedHeadersInSignedOrder(extractedSignedHeaders, pSignValues.SignedHeaders))
```

Comment:
This could simply be `strings.Join(pSignValues.SignedHeaders, ";")` There is no value in re-doing the signed headers while going through the HTTP headers.

Reply:
I thought about this but I wasn't sure if it would be the right method. I will make the changes.

Reply:
Signed headers are already validated, there is no point in regenerating them in the same order.